patches/server/0731-Implement-new-experience-API.patch

```diff
@@ -0,0 +1,50 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: jakebacker <jbacker42@gmail.com>
+Date: Sat, 24 Jul 2021 19:51:06 -0400
+Subject: [PATCH] Implement new experience API
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+index 9e8918d03b8213e5f6689fc93030138fd704aca9..c4286194295844b9fe31444c133b4688dac7a17c 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -1337,6 +1337,39 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+         this.getHandle().totalExperience = exp;
+     }
+ 
++    // Paper start
++    private long getExperienceToNextLevel(int level) {
++        if (level <= 15) {
++            return 2 * level + 7;
++        } else if (level <= 30) {
++            return 5 * level - 38;
++        } else {
++            return 9 * level - 158;
++        }
++    }
```

Comment:
These formulas seems to be wrong, if you see https://minecraft.fandom.com/wiki/Talk:Experience#Formula_for_%22EXP_until_next_level%22_in_incorrect

You should use the vanilla Player#getXpNeededForNextLevel() to be accurate.
Also 3000000 is not a long but still an int which fit into 32 bits signed so give you at least 2^31 exp when you want to test the long usage.